src/App.tsx

```diff
@@ -15,36 +17,154 @@ export const goodsFromServer = [
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: { sortField: string; reverse: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField === SORT_FIELD_NAME) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortField === SORT_FIELD_LENGTH) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+const Button: React.FC<ButtonProps> = ({
+  button,
+  sortField,
+  reverse,
+  setSortField,
+  setReverse,
+}) => {
   return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
+    <button
+      type="button"
+      className={cn(
+        'button',
+        { 'is-info': button === 'Sort alphabetically' },
+        { 'is-success': button === 'Sort by length' },
+        { 'is-warning': button === 'Reverse' },
+        { 'is-danger': button === 'Reset' },
+        {
+          'is-light': button === 'Sort alphabetically' && sortField !== 'name',
+        },
+        { 'is-light': button === 'Sort by length' && sortField !== 'length' },
+        { 'is-light': button === 'Reverse' && !reverse },
+        { 'is-light': button === 'Reset' },
+      )}
+      onClick={() => {
+        const value = button;
+
+        switch (value) {
+          case 'Sort alphabetically':
+            setSortField(SORT_FIELD_NAME);
+
+            break;
+          case 'Sort by length':
+            setSortField(SORT_FIELD_LENGTH);
+            break;
+          case 'Reverse':
+            if (reverse) {
+              setReverse(false);
+            } else {
+              setReverse(true);
+            }
+
+            break;
+          case 'Reset':
+            setSortField('');
+            setReverse(false);
+            break;
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
+          default:
+            break;
+        }
+      }}
+    >
+      {button}
+    </button>
+  );
+};
 
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
+export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverse,
+  });
+  const visibleButtons = getPreparedButtons(buttonsFromServer, {
+    sortField,
+    reverse,
+  });
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {visibleButtons.map((button, index) => (
+          <Button
+            key={index}
```

Comment:
Using the index as a key in React components can lead to issues, especially when the list can change. It's better to use a unique identifier for each button if possible.